lib/ansible/modules/cloud/ovirt/ovirt_vnics_profile.py

```diff
@@ -0,0 +1,254 @@
+    pass_through:
+        description:
+            - "String which enables being directly attached to a VF."
+        choices: ['disabled', 'enabled']
+    migratable:
+        description:
+            - "Boolean which can be set only when pass_through is True."
+        type: bool
+
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+- name: Add vNIC
+  ovirt_vnics_profile:
+    name: myvnic
+    network: mynetwork
+    state: present
+    data_center: datacenter
+
+- name: Editing vNICs network_filter, custom_properties, qos
+  ovirt_vnics_profile:
+    name: myvnic
+    network: mynetwork
+    data_center: datacenter
+    qos: myqos
+    custom_properties:
+      - name: SecurityGroups
+        value: uuid
+    network_filter: allow-dhcp
+
+- name: Editing vNICs network_filter, custom_properties, qos
+  ovirt_vnics_profile:
+    name: myvnic
+    network: mynetwork
+    data_center: datacenter
+    qos: myqos
+    custom_properties:
+      - name: SecurityGroups
+        value: uuid
+    network_filter: allow-dhcp
+
+- name: Dont use migratable
+  ovirt_vnics_profile:
+    name: myvnic
+    network: mynetwork
+    data_center: datacenter
+    migratable: False
+    pass_through: enabled
+
+- name: Remove vNIC
+  ovirt_vnics_profile:
+    name: myvnic
+    network: mynetwork
+    state: absent
+    data_center: datacenter
+'''
+
+RETURN = '''
+id:
+    description: ID of the network interface which is managed
+    returned: On success if network interface is found.
+    type: str
+    sample: 7de90f31-222c-436c-a1ca-7e655bd5b60c
+vnic:
+    description: "Dictionary of all the network interface attributes. Network interface attributes can be found on your oVirt/RHV instance
+                  at following url: http://ovirt.github.io/ovirt-engine-api-model/master/#types/nic."
+    returned: On success if network interface is found.
+    type: dict
+'''
+
+try:
+    import ovirtsdk4.types as otypes
+except ImportError:
+    pass
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ovirt import (
+    BaseModule,
+    check_sdk,
+    create_connection,
+    equal,
+    get_link_name,
+    ovirt_full_argument_spec,
+    search_by_name,
+    get_id_by_name
+)
+
+
+class EntityVnicPorfileModule(BaseModule):
+
+    def __init__(self, *args, **kwargs):
+        super(EntityVnicPorfileModule, self).__init__(*args, **kwargs)
+
+    def __get_network_id(self):
+        dcs_service = self._connection.system_service().data_centers_service()
+        dc_id = get_id_by_name(dcs_service, self.param('data_center'))
+        networks_service = dcs_service.service(dc_id).networks_service()
+        return get_id_by_name(networks_service, self.param('network'))
+
+    def __get_qos_id(self):
+        dcs_service = self._connection.system_service().data_centers_service()
```

Comment:
Maybe you can create method `__get_dc_id()` and reuse it in `__get_network_id` and `__get_qos_id`